lib/Optimizer/Dialect/CC/CCOps.cpp

```diff
@@ -395,6 +432,19 @@ struct FuseCastCascade : public OpRewritePattern<cudaq::cc::CastOp> {
                                                        castToCast.getValue());
         return success();
       }
+      if (isa<IntegerType>(castOp.getType()) &&
+          (castOp.getType() == castToCast.getType())) {
+        // %4 = cc.cast %3 : (i64) -> i64
+        // %5 = cc.cast unsigned %4 : (i64) -> i64
+        // ────────────────────────────────────────────
+        // %5 = cc.cast unsigned %3 : (i64) -> i64
```

Comment:
This is an example of a NOP feeding a NOP. All uses of `%4` and `%5` can be replaced with `%3`.